include/targets/generic/helpers.h

```diff
@@ -0,0 +1,59 @@
+#pragma once
+
+namespace quda
+{
+
+  /**
+     @brief Element type used for coalesced storage.
+   */
+  template <typename T>
+  using atom_t = std::conditional_t<sizeof(T) % 16 == 0, int4, std::conditional_t<sizeof(T) % 8 == 0, int2, int>>;
```

Comment:
Should this be in `target_device.h` since it might be arch dependent?

Reply:
I moved it to load_store.h since that seemed like a good fit, but could move to target_device.h (and add target:: or device::) if preferred.